docs/lib/index.rst

```diff
@@ -0,0 +1,221 @@
+
+    :ref:`Server-side sessions <lib/usage/middleware/builtin-middleware:memcached storage>` / :ref:`redis caching <lib/usage/caching:memcached>`:
+        :code:`pip install starlite[memcached]`
+
+    :ref:`Picologging <lib/usage/the-starlite-app:using picologging>`
+        :code:`pip install starlite[picologging]`
+
+    :ref:`StructLog <lib/usage/the-starlite-app:using structlog>`
+        :code:`pip install starlite[structlog]`
+
+    :doc:`/lib/usage/contrib/open-telemetry`
+        :code:`pip install starlite[openetelemetry]`
+
+    :doc:`/lib/usage/cli`
+        :code:`pip install starlite[cli]`
+
+    Standard installation (includes CLI, picologging and Jinja2 templating):
+        :code:`pip install starlite[standard]`
+
+    All extras:
+        :code:`pip install starlite[full]`
+
+
+
+Minimal Example
+---------------
+
+**Define your data model** using pydantic or any library based on it (for example ormar, beanie, SQLModel):
+
+.. code-block:: python
+
+    from pydantic import BaseModel, UUID4
+
+
+    class User(BaseModel):
+        first_name: str
+        last_name: str
+        id: UUID4
+
+
+
+
+You can also use dataclasses (standard library and Pydantic),
+:class:`typing.TypedDict` or :class:`msgspec.Struct`.
+
+.. code-block:: python
+
+   from uuid import UUID
+
+   # from pydantic.dataclasses import dataclass
+   from dataclasses import dataclass
+
+
+   @dataclass
+   class User:
+       first_name: str
+       last_name: str
+       id: UUID
+
+**Define a Controller for your data model:**
+
+.. code-block:: python
+
+    from typing import List
+
+    from pydantic import UUID4
+    from starlite import Controller, Partial, get, post, put, patch, delete
+
+    from my_app.models import User
+
+
+    class UserController(Controller):
+        path = "/users"
+
+        @post()
+        async def create_user(self, data: User) -> User:
+            ...
+
+        @get()
+        async def list_users(self) -> List[User]:
+            ...
+
+        @patch(path="/{user_id:uuid}")
+        async def partial_update_user(self, user_id: UUID4, data: Partial[User]) -> User:
+            ...
+
+        @put(path="/{user_id:uuid}")
+        async def update_user(self, user_id: UUID4, data: User) -> User:
+            ...
+
+        @get(path="/{user_id:uuid}")
+        async def get_user(self, user_id: UUID4) -> User:
+            ...
+
+        @delete(path="/{user_id:uuid}")
+        async def delete_user(self, user_id: UUID4) -> None:
+            ...
+
+
+When instantiating your app, import your *controller* into your application's
+entry-point and pass it to Starlite:
+
+.. code-block:: python
+
+   from starlite import Starlite
+
+   from my_app.controllers.user import UserController
+
+   app = Starlite(route_handlers=[UserController])
+
+To **run your application**, use an ASGI server such as `uvicorn <https://www.uvicorn.org/>`_ :
+
+.. code-block:: shell
+
+   uvicorn my_app.main:app --reload
+
+
+Philosophy
```

Comment:
I suggest this section is split into a different page and rewritten .

Reply:
@Jarmos-san is working on that in #1153